src/_pytest/nodes.py

```diff
@@ -44,46 +43,39 @@
 tracebackcutdir = py.path.local(_pytest.__file__).dirpath()
 
 
-@lru_cache(maxsize=None)
-def _splitnode(nodeid: str) -> Tuple[str, ...]:
-    """Split a nodeid into constituent 'parts'.
+def getparentnodeids(nodeid: str) -> Iterator[str]:
```

Comment:
Nice function and docs!

Did you try to measure with `@lru_cache` being applied to it?

Reply:
No, but I think it shouldn't be a hot spot anymore, and shouldn't be called with the same nodeid a million times like before. But if it does show up in a profile again it might make sense.